File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int sz = nums.size(), ans = 0;
        for(int i=0; i<sz/2; i++){
            ans = max(ans, nums[i] + nums[sz-1-i]);
        }
        return ans;
    }
};

File: 1988-minimize-maximum-pair-sum-in-array/minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        int sz = nums.length, ans = -1;
        Arrays.sort(nums);

        for(int i=0; i<sz/2; i++){
            ans = ans > (nums[sz - 1 - i] + nums[i]) ? ans : (nums[sz - 1 - i] + nums[i]);
        }

        return ans;
    }
}